### PR TITLE
added the grocoder permission

### PR DESCRIPTION
**Pull request by adil-uddin**
https://dimagi-dev.atlassian.net/browse/USH-209

##### SUMMARY

Changing the default geocoded location label and putting the setting field behind the subscription for web apps.

### CODE REVIEW

**Comment by adil-uddin** [comment]
Without Webapps privilege:
Screenshot

With Webapps privilege:
Screenshot1

@orangejenny please review.

File: corehq/apps/domain/forms.py [hunk 1]
```diff
@@ -537,6 +537,9 @@ def __init__(self, *args, **kwargs):
             # if the cloudcare_releases flag was just defaulted, don't bother showing
             # this setting at all
             del self.fields['cloudcare_releases']
+        if self.project.default_geocoder_location == 'default' \
```
Comment (orangejenny) [inline corehq/apps/domain/forms.py:540]: What does the check for `default` do?

File: corehq/apps/domain/forms.py [hunk 2]
```diff
@@ -537,6 +537,9 @@ def __init__(self, *args, **kwargs):
...
+                or not domain_has_privilege(self.domain, privileges.CLOUDCARE):
+            del self.fields['default_geocoder_location']
```
Comment (orangejenny) [inline corehq/apps/domain/forms.py:542]: I think you also need to update the code in `DomainGlobalSettingsForm.__init__` that's based on the index of different fields: https://github.com/dimagi/commcare-hq/blob/64d433902599bbfa56261047ff3d4a8cdf9af38f/corehq/apps/domain/forms.py#L392-L393

I think that `__init__` code is what's causing the logo field to not appear, as discussed in the USH Tech chat on Skype.

File: corehq/apps/domain/forms.py [hunk 3]
```diff
@@ -389,8 +389,10 @@ def __init__(self, *args, **kwargs):
         self.can_use_custom_logo = kwargs.pop('can_use_custom_logo', False)
         super(DomainGlobalSettingsForm, self).__init__(*args, **kwargs)
         self.helper = hqcrispy.HQFormHelper(self)
-        self.helper[4] = twbscrispy.PrependedText('delete_logo', '')
-        self.helper[5] = twbscrispy.PrependedText('call_center_enabled', '')
+        self.helper[3] = twbscrispy.PrependedText('default_geocoder_location', '')
```
Comment (orangejenny) [inline corehq/apps/domain/forms.py:392]: Does this work regardless both when the user has `privileges.CLOUDCARE` and when they do not?

Comment (adil-uddin) [reply corehq/apps/domain/forms.py]: reverted as this change is not needed. But it works in both cases.

File: corehq/apps/domain/forms.py (hunk 2 above)
Comment (adil-uddin) [reply corehq/apps/domain/forms.py:541]: Changed the indexes and tested. This fixed the upload logo in my local

File: corehq/apps/domain/forms.py (hunk 1 above)
Comment (adil-uddin) [reply corehq/apps/domain/forms.py]: turns out it was not needed

**Review by orangejenny — APPROVED** [review]

**Pull request merged**